### DISCUSSION TITLE
[Bug]:  ESP32 "Light Sleep" doesn't work if device is in the CLIENT role

### DISCUSSION THREAD

**Discussion by h0lad** [meshtastic/firmware#5078] [Troubleshooting]
### Category

Other

### Hardware

Heltec V3

### Firmware Version

2.5.x

### Description

If the device is set as CLIENT with power saving activated it stays awake the entire time.

How to reproduce

1. Set "CLIENT" role
2. Set "Minimum Wake Interval" to 1s
3. Look at the console

It stays awake the entire time:

```
DEBUG | ??:??:?? 662 [Power] Battery: usbPower=0, isCharging=0, batMv=4122, batPct=95
DEBUG | ??:??:?? 682 [Power] Battery: usbPower=0, isCharging=0, batMv=4122, batPct=95
DEBUG | ??:??:?? 702 [Power] Battery: usbPower=0, isCharging=0, batMv=4122, batPct=95
DEBUG | ??:??:?? 722 [Power] Battery: usbPower=0, isCharging=0, batMv=4122, batPct=95
```

Compared to the ROUTER role:

```
INFO  | ??:??:?? 1 [RangeTestModule] Range Test Module - Disabled
DEBUG | ??:??:?? 4 [PowerFSM] Enter state: ON
INFO  | ??:??:?? 7 [PowerFSM] lsEnter begin, ls_secs=86400
INFO  | ??:??:?? 7 [PowerFSM] lsEnter end
INFO  | ??:??:?? 7 [PowerFSM] setup LORA_DIO1 (GPIO03) with wakeup by gpio interrupt
INFO  | ??:??:?? 37 [PowerFSM] Exit light sleep cause: 4
INFO  | ??:??:?? 37 [PowerFSM] setup LORA_DIO1 (GPIO03) with wakeup by gpio interrupt
INFO  | ??:??:?? 37 [PowerFSM] Exit light sleep cause: 4
INFO  | ??:??:?? 37 [PowerFSM] sleeping, flash led!
(...)
```

**Comment by thebentern** [comment]
screen_on_secs and wait_bluetooth_secs also delay Light Sleep

**Reply by h0lad** [reply]
Can confirm. Reducing both to 1s fixed the issue. Device now properly sleeps in "CLIENT" role. 

Thanks for super fast help.

**Reply by thebentern** [reply]
No worries. The power FSM is a complex beast. Router mode cuts through some of the states adding to confusion. One of these days we'll do some cleanup and consolidation.

**Comment by dongdigua** [comment]
https://github.com/meshtastic/firmware/blob/29893e0c281c7266eeeffe956a5b1c367dfc9417/src/mesh/PhoneAPI.cpp#L278

is this line ignoring my setting???

whatever value I set, it's always 300 on CLIENT
